lib/adf_diag.py

```diff
@@ -1080,3 +1108,227 @@ def derive_variables(self, vars_to_derive=None, ts_dir=None, overwrite=None):
+                                "ERROR: None of the frequency options {freq_string_options} are present in the time_period_freq attribute {time_perido_freq}"
+                            )
+                            print("Skipping {adf_file}")
+                            freq = "frequency_missing"
+                        continue
+                    freq = found_strings[0]
+
+                    #
+                    # Destination file is MDTF directory and name structure
+                    #
+                    mdtf_dir = os.sep.join([mdtf_model_data_root, case_name, freq])
+
+                    os.makedirs(mdtf_dir, exist_ok=True)
+                    mdtf_file = (
+                        mdtf_dir + os.sep + ".".join([case_name, var, freq, "nc"])
+                    )
+                    mdtf_file_list = glob.glob(
+                        mdtf_file
+                    )  # Check if files already exist in MDTF directory
+                    if (
+                        mdtf_file_list
+                    ):  # If files exist, then should check if over-writing is allowed:
+                        if verbose > 0:
+                            print(
+                                f"\t   INFO: not clobbering existing mdtf file {mdtf_file_list}"
+                            )
+                        # logic to allow override#  if not overwrite_mdtf[case_idx]:
+                        continue  # simply skip file copy for this variable:
+
+                    if verbose > 1:
+                        print(f"copying {adf_file} to {mdtf_file}")
+                    shutil.copyfile(adf_file, mdtf_file)
+                # end for hist_str
+            # end for var
+        # end for case
+
+    def setup_run_mdtf(self):
+        """
+        Create MDTF directory tree, generate input setttings jsonc file
+        Submit MDTF diagnostics.
+
+        """
+
+        # import needed standard modules:
+        import json
+
+        test = False  # True (copy files but don't run), False (copy files and run MDTF).  Could make this a yaml variable
+        verbose = 2  # 0 errors and major sections, 1 warnings, 2 detailed, 3 excessive
+
+        print("\n  Setting up MDTF...")
+        # We want access to the entire dict of mdtf_info
+        mdtf_info = self.__mdtf_info
+
+        #
+        # Create a dict with all the case info needed for MDTF case_list
+        #     Note that model and convention are hard-coded to CESM because that's all we expect here
+        #     This could be changed by inputing them into ADF with other MDTF-specific variables
+        #
+        case_list_keys = ["CASENAME", "FIRSTYR", "LASTYR", "model", "convention"]
+
+        # Casenames, paths and start/end years come through the ADF
+        case_names = self.get_cam_info("cam_case_name", required=True)
+        start_years = self.climo_yrs["syears"]
+        end_years = self.climo_yrs["eyears"]
+
+        case_list_all = list(())
```

Comment:
Too many parentheses here, also better to just use the symbolic list representation:

```suggestion
        case_list_all = []
```